client/src/pages/trainer/courses/create/index.tsx

```diff
@@ -58,15 +60,22 @@ const Create: FC = () => {
         }
         return await trigger(['name', 'category']);
       case 2:
-        /*
-            Please collect the form inputs in here, since user is at the preview section, crouseSlice should have all the necessary info we need to save in db
-
-            Please reset the courseSlice and stepperSlice to initial values once the from values are saved in db
-        */
-        return await push('/trainer/courses');
+        try {
+          const data = {
+            ...values,
+            category: values.category.map(({ id }) => id),
+          };
+          await API.post('course/', data);
+          void push('/trainer/courses');
+          dispatch(reset());
+        } catch (error) {
+          console.error('Error saving course data:', error);
+        }
+        break;
       default:
         return validateSteps();
     }
+    return false;
```

Comment:
May I know what you need this return for? 🙇‍♂️

Reply:
There is  an error at first in the Promise<boolean>  that says "Function lacks ending return statement and return type does not include 'undefined'.", so I put a return false.